docs/enriching-your-data/available-enrichments/custom-javascript-enrichment/examples/index.md

```diff
@@ -0,0 +1,95 @@
+---
+title: "Example code"
+sidebar_position: 3
+description: "Examples of enrichment code in JavaScript."
+---
+
+## Labeling internal and external traffic
+
+Suppose you want to extend your events with a label that distinguishes traffic from internal and external systems. This is a good use case for [adding an entity](/docs/enriching-your-data/available-enrichments/custom-javascript-enrichment/writing/index.md#adding-extra-entities-to-the-event).
+
+```js
+const internalIPs = new Set([
+  "1.2.3.4",
+  "2.3.4.5"
+]);
+
+function process(event) {
+  const ip = event.getUser_ipaddress();
+  const traffic_source = internalIPs.has(ip) ? 'internal' : 'external';
+
+  return [{
+    schema: "iglu:com.my-company/traffic_source/jsonschema/1-0-0",
+    data: { traffic_source }
+  }];
+}
+```
+
+:::note
+
+This example is not copy-paste-able. You will need to define your own schema.
+
+:::
+
+## Filtering out bots
+
+You can use the [IAB enrichment](/docs/enriching-your-data/available-enrichments/iab-enrichment/index.md) to identify bots. However, it does not remove the bot-generated events from your data — it only adds relevant information to the events, e.g.:
+
+```json
+{
+    "schema": "iglu:com.iab.snowplow/spiders_and_robots/jsonschema/1-0-0",
+    "data": {
+        "spiderOrRobot": false,
+        "category": "BROWSER",
+        "reason": "PASSED_ALL",
+        "primaryImpact": "NONE"
+    }
+}
+```
+
+If you wish to discard the bot events (sending them to [failed events](/docs/managing-data-quality/failed-events/understanding-failed-events/index.md)), you can use the JavaScript enrichment:
+
+```js
+function process(event) {
+  const entities = JSON.parse(event.getContexts());
+  if (entities) {
+    for (const entity of entities.data) {
+      if (entity.schema === 'iglu:com.iab.snowplow/spiders_and_robots/jsonschema/1-0-0' &&
+          entity.data.spiderOrRobot) {
+        throw "Filtered a spider/bot event";
+      }
+    }
+  }
+  return [];
+}
+```
+
+For special cases, you can also dispense with the IAB enrichment and add something simpler:
+
+```js
+const botPattern = /.*Googlebot.*/;
+
+function process(event) {
+  const useragent = event.getUseragent();
+
+  if (useragent !== null && botPattern.test(useragent)) {
+    throw "Filtered event produced by Googlebot";
+  }
+
+  return [];
+}
+```
+
+## Eliding information based on a condition
+
+Let’s say you want to elide certain user information depending on which country the user is in. While [PII pseudonymization enrichment](/docs/enriching-your-data/available-enrichments/pii-pseudonymization-enrichment/index.md) is very useful for dealing with PII, it does not support such conditions. JavaScript enrichment to the rescue!
```

Comment:
Elide (apparently, I had never heard of it) means to omit while speaking. Maybe something like redact, remove, obfuscate, omit, destroy,...